packages/react-devtools-shell/package.json

```diff
@@ -24,10 +24,11 @@
     "babel-loader": "^8.0.4",
     "cross-env": "^3.1.4",
     "css-loader": "^1.0.1",
+    "html-webpack-plugin": "^5",
     "raw-loader": "^3.1.0",
     "style-loader": "^0.23.1",
-    "webpack": "^4.43.0",
-    "webpack-cli": "^3.3.11",
-    "webpack-dev-server": "^3.10.3"
+    "webpack": "^5",
+    "webpack-cli": "^4",
+    "webpack-dev-server": "^4"
   }
 }
```

Comment:
The `react-devtools-shell` test app is broken by this change. (I think actually it's the `react-devtools-inline` package that gets broken, but I'll explain that below.)

Looks like the page is trying to request `/dist/devtools.js` and it's getting a 400 response. (Interestingly, the page is also requesting `/devtools.js` and that's returning okay.)

Editing the `index.html` file so the `<script>` tag loads `/devtools.js` instead (no `/dist/...`) fixes the failed network requests, but still doesn't fix the broken shell.

I see the following error in the console:
```
Uncaught TypeError: (0 , react_devtools_inline_backend__WEBPACK_IMPORTED_MODULE_2__.initialize) is not a function
```

I also see a few (related?) warnings in the console when running the test shell:
```
WARNING in ./src/devtools.js 14:0-17
export 'initialize' (imported as 'initializeBackend') was not found in 'react-devtools-inline/backend' (possible exports: backend)

WARNING in ./src/devtools.js 17:17-35
export 'initialize' (imported as 'initializeFrontend') was not found in 'react-devtools-inline/frontend' (possible exports: frontend)

WARNING in ./src/devtools.js 20:0-15
export 'activate' (imported as 'activateBackend') was not found in 'react-devtools-inline/backend' (possible exports: backend)
```

Inspecting the imported module, it looks like there's an extra `backend` wrapper now, e.g.
```js
// module:
{
  backend: {
    activate: (...),
    initialize: (...)
  }
}
```

Whereas before it was:
```js
// module:
{
  activate: (...),
  initialize: (...)
}
```

This behavior can be reproduced by following the steps here:
https://github.com/facebook/react/blob/main/packages/react-devtools/CONTRIBUTING.md#option-1-fastest-using-the-test-shell

Then loading localhost:8080.

Reply:
I think we _could_ work around this by updating the entrypoint files like so:

#### react-devtools-inline/backend.js
```js
module.exports = require('./dist/backend').backend;
```

#### react-devtools-inline/frontend.js
```js
module.exports = require('./dist/frontend').frontend;
```

But I don't know why this is necessary, and what other places it might bite us?

Reply:
Looks like I can fix the GET failures by modifying the following files:
```diff
diff --git a/packages/react-devtools-shell/index.html b/packages/react-devtools-shell/index.html
index 410dc5bdc..732ba5db6 100644
--- a/packages/react-devtools-shell/index.html
+++ b/packages/react-devtools-shell/index.html
@@ -64,6 +64,6 @@
     <!-- This script installs the hook, injects the backend, and renders the DevTools UI -->
     <!-- In DEV mode, this file is served by the Webpack dev server -->
     <!-- For production builds, it's built by Webpack and uploaded from the local file system -->
-    <script src="dist/devtools.js"></script>
+    <script src="devtools.js"></script>
   </body>
 </html>
\ No newline at end of file

diff --git a/packages/react-devtools-shell/src/devtools.js b/packages/react-devtools-shell/src/devtools.js
index fda4c91b7..c7fbd3a1c 100644
--- a/packages/react-devtools-shell/src/devtools.js
+++ b/packages/react-devtools-shell/src/devtools.js
@@ -50,7 +50,7 @@ mountButton.addEventListener('click', function() {
   }
 });
 
-inject('dist/app.js', () => {
+inject('app.js', () => {
   initDevTools({
     connect(cb) {
       const root = createRoot(container);
```

But now for some reason the test app code loads 3 times in the page (so there's 3 duplicate copies) and DevTools UI is still not visible at all (though it's loading). It looks like the CSS/styles are weird somehow, because the DevTools UI is loading in the background (behind the duplicate copies of the app).

Digging in a little more, I see a lot of `<div>` elements in the root of the page (outside of the `<iframe>` the test app is supposed to load into). Deleting them reveals the DevTools UI underneath, and it even mostly works, but there are a lot of "duplicate ID" errors in the console (presumably from the duplicate test apps) so...something wonky is still going on here.

Reply:
Possibly related to the "3 copies of the app are on the page" thing I mentioned above– the console also shows 3 copies of these logs:
```
VM7176 devtools.js:17483         [HMR] Waiting for update signal from WDS...
VM7176 devtools.js:4692          Download the React DevTools for a better development experience: https://reactjs.org/link/react-devtools
log.js:24                        [HMR] Waiting for update signal from WDS...
react-dom.development.js:29044   Download the React DevTools for a better development experience: https://reactjs.org/link/react-devtools
console.js:29                    [HMR] Waiting for update signal from WDS...
react-dom.development.js:29044   Download the React DevTools for a better development experience: https://reactjs.org/link/react-devtools
VM7176 devtools.js:11398         [webpack-dev-server] Hot Module Replacement enabled.
index.js:519                     [webpack-dev-server] Hot Module Replacement enabled.
index.js:519                     [webpack-dev-server] Hot Module Replacement enabled.
VM7176 devtools.js:11398         [webpack-dev-server] Live Reloading enabled.
index.js:519                     [webpack-dev-server] Live Reloading enabled.
index.js:519                     [webpack-dev-server] Live Reloading enabled.
```

Spaces added by me to highlight the fact that– although the content of the logs is the same, the file locations are different. I think this suggests that Webpack is bundling things oddly?